lib/js/index.js

```diff
@@ -91,10 +95,13 @@ class DateTimePicker extends Events {
       //  and deal with updating the view only).
       // For now this allows us to set the default time using the same quantize
       // rules as setting the date explicitly. Setting this.value meets setTime|Date's
-      // expectation that we have a value, and `0` guarantees that we will detect 
+      // expectation that we have a value, and `0` guarantees that we will detect
       this.value = moment(0);
       this.setDate(this.options.default);
       this.setTime(this.options.default);
+    } else {
+      this.setDate(this.value);
```

Comment:
Thanks for this fix 👍 

As per the `TODO` this methods needs refactoring, because this type of bug is quite common! I hope to get around to that soon.